Add docs for List.take, List.takeWhile, List.sub (#122)
grain-lang/grain-lang.org: src/stdlib/list.md

Returns the index of the first item that the predicate returns `true` for.
Fails if an item is not found.

### List.**sub**

```grain
sub: (Number, Number, List<a>) -> List<a>
```

> Returns a sub-list of a given list staring from index `start` (inclusive) and with lenght `length`.
> _Note that length of the resulting list can be less then `length` parameter in case of source list not having enought elements._

Reviewer: ```suggestion
Returns a sub-list of the given list starting from index `start` (inclusive) and up to `length` elements.
```
